src/components/MoneyRequestConfirmationList.tsx

```diff
@@ -413,43 +401,99 @@ function MoneyRequestConfirmationList({
+                }
+                return {
+                    ...participantOption,
+                    descriptiveText: amount ? CurrencyUtils.convertToDisplayString(amount, iouCurrencyCode) : '',
+                };
+            });
+        }
+
+        const currencySymbol = currencyList?.[iouCurrencyCode ?? '']?.symbol ?? iouCurrencyCode;
+        return [payeeOption, ...selectedParticipants].map((participantOption: Participant) => ({
+            ...participantOption,
+            tabIndex: -1,
+            amountInputOptions: {
+                amount: transaction?.splitShares?.[participantOption.accountID ?? 0]?.amount,
+                currency: iouCurrencyCode,
+                prefixCharacter: currencySymbol,
+                isCurrencyPressable: false,
+                hideCurrencySymbol: true,
+                textInputContainerStyles: [{minWidth: 50}],
+                onAmountChange: (value: string) => onSplitShareChange(participantOption.accountID ?? 0, Number(value)),
+            },
+        }));
+    }, [transaction, iouCurrencyCode, isPolicyExpenseChat, onSplitShareChange, payeePersonalDetails, selectedParticipants, currencyList, iouAmount, isReadOnly]);
+
     const optionSelectorSections = useMemo(() => {
         const sections = [];
-        const unselectedParticipants = selectedParticipantsProp.filter((participant) => !participant.selected);
         if (hasMultipleParticipants) {
-            const formattedSelectedParticipants = getParticipantsWithAmount(selectedParticipants);
-            let formattedParticipantsList = [...new Set([...formattedSelectedParticipants, ...unselectedParticipants])];
-
-            if (!canModifyParticipants) {
-                formattedParticipantsList = formattedParticipantsList.map((participant) => ({
-                    ...participant,
-                    isDisabled: ReportUtils.isOptimisticPersonalDetail(participant.accountID ?? -1),
-                }));
-            }
-
-            const myIOUAmount = IOUUtils.calculateAmount(selectedParticipants.length, iouAmount, iouCurrencyCode ?? '', true);
-            const formattedPayeeOption = OptionsListUtils.getIOUConfirmationOptionsFromPayeePersonalDetail(
-                payeePersonalDetails,
-                iouAmount > 0 ? CurrencyUtils.convertToDisplayString(myIOUAmount, iouCurrencyCode) : '',
-            );
-
-            sections.push(
-                {
-                    title: translate('moneyRequestConfirmationList.paidBy'),
-                    data: [formattedPayeeOption],
-                    shouldShow: true,
-                    isDisabled: shouldDisablePaidBySection,
-                },
-                {
-                    title: translate('moneyRequestConfirmationList.splitWith'),
-                    data: formattedParticipantsList,
-                    shouldShow: true,
-                },
-            );
+            const formattedParticipantsList = getParticipantOptions();
+            sections.push({
```

Comment:
![Screenshot 2024-04-23 at 1 48 15 PM](https://github.com/Expensify/App/assets/102477862/1c8f0ac1-dd5c-49b5-919c-73652572d2dd)

This PR on left. `main` on right.

'Paid by' section is removed. Is this intentional?

Could you share the section of the document for this feature on kkch3024@gmail.com if it is allowed? I have signed the NDA.

The amounts are missing for some participants.

Reply:
>Also, the amounts are missing for some participants.

Unable to reproduce this one either.

Reply:
Actually I can now with Hidden accounts

Reply:
This will be fixed in the backend. Because the accountIDs used in that page are real accountIDs which should come back from the HTTPS response

Reply:
>'Paid by' section is removed. Is this intentional?

Yes, it will be brought back with this [PR](https://github.com/Expensify/App/pull/40388)